dapps/settings/src/components/identity/side-panel.ts

```diff
@@ -0,0 +1,213 @@
+   */
+  async save(): Promise<void> {
+    const runtime = this.getRuntime();
+    const contact = {
+      ...this.contact,
+      ...this.form.getFormData(),
+    };
+
+    // start identity invite dispatcher and pass translated b-mail content
+    const fromAlias = await profileUtils.getUserAlias(runtime);
+    identityShareDispatcher.start(this.getRuntime(), {
+      contact,
+      bmail: {
+        fromAlias,
+        title: this.$t('_settings.identity.share.title'),
+        body: this.$t('_settings.identity.share.body', {
+          alias: fromAlias,
+          identity: runtime.activeIdentity,
+          access: this.$t(`_settings.identity.share.type.${contact.hasIdentityAccess}`),
+        }).replace(/\n/g, '<br>'),
+      },
+    });
+
+    this.hie();
+  }
+
+  /**
+   * Setup the edit formular.
+   */
+  setupForm(): void {
+    this.form = (new EvanForm(this, {
+      address: {
+        value: this.contact.address || '',
+        validate: (vueInstance, form, control: EvanFormControl): boolean|string => session
+          .identityRuntime.web3.utils.isAddress(control.value),
+        uiSpecs: {
+          type: 'v-select',
+          attr: {
+            required: true,
+            options: this.contacts.map((contact: IdentityAccessContact) => ({
+              value: contact.address,
+              label: contact.displayName || contact.address,
+            })),
+          },
+          input: (address: string) => {
+            if (address) {
+              this.show(this.contacts.find((contact) => contact.address === address));
+            }
+          },
+        },
+      },
+      note: {
+        value: this.contact.note || '',
+      },
+    }) as IdentityForm);
+  }
+
+  /**
+   * Update the permissions object to match the hasIdentityAccess param.
+   */
+  setupPermissions(): void {
+    this.permissions = {
+      identity: {
+        read: this.contact.hasIdentityAccess === 'read' || this.contact.hasIdentityAccess === 'write',
+        readWrite: this.contact.hasIdentityAccess === 'write',
+      },
+    };
+  }
+
+  /**
+   * Move the side-panel in.
+   */
+  show(contact: IdentityAccessContact): void {
+    this.originalContact = contact || { } as IdentityAccessContact;
+    this.contact = { ...this.originalContact };
+    this.isNew = !contact;
+    this.setupForm();
+    this.setupPermissions();
+    (this.$refs.sidePanel as SwipePanelComponentClass).show();
+  }
+
+  /**
+   * Write permission changes to current contact
+   *
+   * @param      {PermissionsInterface}  permissions  latest permission confugration
+   */
+  updatePermissions({ permissions }: { permissions: PermissionsInterface }): void {
+    const idPerm = permissions.identity;
+    const originPerm = this.originalContact.hasIdentityAccess;
+
+    // do not allow to remove read permissions
+    // eslint-disable-next-line no-nested-ternary
+    let resolvedPerm = idPerm.readWrite ? 'write' : idPerm.read ? 'read' : false;
```

Comment:
I wouldn't ignore the rule here. This is hard to read. Rather use a normal if statement :)